WcaOnRails/app/models/competition.rb

```diff
@@ -755,8 +763,14 @@ def events_to_associated_events(events)
     end
   end
 
-  def self.search(query, params: {})
-    competitions = Competition.where(showAtAll: true)
+  def self.search(query, params: {}, current_resource_owner: nil)
+    if params[:managed_by_me].present?
+      raise WcaExceptions::BadApiParameter.new("Must be logged in to specify `managed_by_me`") unless current_resource_owner
+      raise WcaExceptions::BadApiParameter.new("Missing required scope `manage_competitions`") unless current_resource_owner.doorkeeper_token.scopes.include?("manage_competitions")
```

Comment:
Same here, could we have some kind of redirect to a common "not logged/authorized" error view?
(really just asking, I don't know what's usually done!)